tests/_support/Helper/Acceptance.php

```diff
@@ -320,7 +320,12 @@ public function wooCommerceCreateProductAndCheckoutWithConfig(
 
 		// Define Send Purchase Data setting.
 		if ($sendPurchaseData) {
-			$I->checkOption('#woocommerce_ckwc_send_purchases');	
+			$I->checkOption('#woocommerce_ckwc_send_purchases');
+
+			// If sendPurchaseData is true, set send purchase data event to processing.
+			// Otherwise set to the string value of sendPurchaseData i.e. completed.
+			$sendPurchaseDataEvent = (($sendPurchaseData === true) ? 'processing' : $sendPurchaseData);
```

Comment:
Pardon my ignorance, but can't this just be:
`$sendPurchaseDataEvent = $sendPurchaseData ? 'processing' : $sendPurchaseData;`

Reply:
> Nice! Love the tests, and wondering if we should add one that toggles the settings first and then makes sure that the plugin adjusts its behavior accordingly.
- `PurchaseDataCest:testSendPurchaseDataOnOrderCompletedWithSimpleProductCheckout` tests when setting = Order Completed.
- Other tests in PurchaseDataCest test when setting = Order Processing. 
- `SettingPurchasesCest:testSendPurchaseDataEventOrderProcessing` and `SettingPurchasesCest:testSendPurchaseDataEventOrderCompleted` added to confirm WooCommerce saves changes to this setting.
- Both settings therefore covered.

> Pardon my ignorance, but can't this just be: `$sendPurchaseDataEvent = $sendPurchaseData ? 'processing' : $sendPurchaseData;`
- `$sendPurchaseData` is of mixed type (string or bool), therefore fine (and good practice) to use strict comparison operator to ensure value **and** type match (i.e. boolean true), not just value (i.e. string '1' which would evaluate as true also).